### ISSUE TITLE
Type safe rule disable non-type safe version?

### ISSUE THREAD

**Issue by OmgImAlexis** [typescript-eslint/typescript-eslint#1853]
**What were you trying to do?**
I want to have my extended config support js and ts.

```json
{
  "rules": {
    "@typescript-eslint/space-before-function-paren": [
      "error",
      "never"
    ],
    "space-before-function-paren": [
      "error",
      "never"
    ]
  }
}
```

```ts
// Put your code here
```

**What did you expect to happen?**
I had hoped since Typescript has the `typescript-eslint` versions of some rules it'd disable the existing ones if they matched in name.

**What actually happened?**
```
  ✖  290:32  Unexpected space before function parentheses.  @typescript-eslint/space-before-function-paren
  ✖  290:32  Unexpected space before function parentheses.  space-before-function-paren
```

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `2.26.0` |
| `@typescript-eslint/parser`             | `2.26.0 ` |
| `@typescript-eslint/typescript-estree`  | `2.26.0` |
| `@typescript-eslint/experimental-utils` | `2.26.0` |
| `TypeScript`                            | `3.8.3` |
| `node`                                  | `v12.5.0` |
| `npm`                                   | `6.13.0` |

**Comment by bradzacher** [comment]
You should use ESLint's overrides functionality to configure rules based on globs
https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns

**Comment by OmgImAlexis** [comment]
I’ve seen other ESLint rule packages do exactly what I’m staying. I get ESLint itself doesn’t have this ability but I’m sure there’s a way to bake this in.

**Comment by bradzacher** [comment]
> I’ve seen other ESLint rule packages do exactly what I’m staying

Have you got examples? I'd be interested in seeing them.

The closest thing that a plugin can do is to provide premade configs, which we do already:
https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/configs
